group-actors/src/main/java/org/sunbird/service/impl/GroupServiceImpl.java

```diff
@@ -82,15 +85,39 @@ public String createGroup(Group groupObj) throws BaseException {
                 memberService.fetchMembersByGroupIds(Lists.newArrayList(groupId), null);
             dbResGroup.put(JsonKey.MEMBERS, members);
           }
-        }
-        if (CollectionUtils.isEmpty(fields) || !fields.contains(JsonKey.ACTIVITIES)) {
-          dbResGroup.remove(JsonKey.ACTIVITIES);
+          if (fields.contains(JsonKey.ACTIVITIES)) {
+            List<Map<String, Object>> dbResActivities =
+                (List<Map<String, Object>>) dbResGroup.get(JsonKey.ACTIVITIES);
+            if (dbResActivities != null && !dbResActivities.isEmpty()) {
+              addActivityInfoDetails(dbResActivities);
+            }
+          } else {
+            removeActivities(dbResGroup);
+          }
+        } else {
+          removeActivities(dbResGroup);
```

Comment:
@sknirmalkar89 - dont we need to remove members also from the dbResGroup if "fields" is empty?

Reply:
not Required since member information is stored in different table and will only be fetched when fields contains members